src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1134,56 +1188,117 @@ private void RunLayoutGraph(GraphLayout.Graph graph)
             graph.SetGraphPosition();
         }
 
-        private void SaveLayoutGraph(GraphLayout.Graph graph)
+        private void AvoidSubgraphOverlap()
+        {
+            var VerticalGraphDistance = 30;
+            var HorizontalGraphDistance = 70;
+
+            bool done;
+
+            do
+            {
+                done = true;
+
+                foreach (var g1 in Model.LayoutSubgraphs.Skip(1))
+                {
+                    foreach (var g2 in Model.LayoutSubgraphs.Skip(1))
+                    {
+                        if (!g1.Equals(g2) && (g1.GraphCenterY + g1.OffsetY <= g2.GraphCenterY + g2.OffsetY))
+                        {
+                            var g1nodes = g1.Nodes.OrderBy(n => n.Y + n.Height);
+                            var g2nodes = g2.Nodes.OrderBy(n => n.Y);
+
+                            foreach (var node1 in g1nodes)
+                            {
+                                foreach (var node2 in g2nodes)
+                                {
+                                    if ((node1.Y + node1.Height + VerticalGraphDistance + g1.OffsetY > node2.Y + g2.OffsetY) &&
```

Comment:
Similarly you could add a comment here to describe what you're checking for here in this complex `if` condition.